### PR TITLE
enable use of custom data-test-id

### PR DESCRIPTION
**Pull request by shahzad31**
With this we can provide a custom data-test id like `data-test-subj`, will make life easier with kibana tests specifically.

### CODE REVIEW

File: __tests__/core/gatherer.test.ts [hunk 1]
```diff
@@ -148,7 +148,7 @@ describe('Gatherer', () => {
         expect(request.headers()['user-agent']).toContain('Elastic/Synthetics');
       });
       await page.setContent(
-        '<a target=_blank rel=noopener href="/popup.html">popup</a>'
+        '<a target=_blank rel=noopener href="/username-page">popup</a>'
```
Comment (shahzad31) [inline __tests__/core/gatherer.test.ts:151]:
```suggestion
        '<a target=_blank rel=noopener href="/popup.html">popup</a>'
```

**Review by vigneshshanmugam — COMMENTED** [review]
@shahzad31 Whats the motivation behind this change? I have couple of questions on this

1. why cant Kibana set these attributes on `beforeAll` hook? 
2. Also, since you are doing this on the overall journey, How would it help in the Kibana context since the whole journey could contain multiple elements with these test attributes?

**Comment by shahzad31** [comment]
> @shahzad31 Whats the motivation behind this change? I have couple of questions on this
> 
> 1. why cant Kibana set these attributes on `beforeAll` hook?
> 2. Also, since you are doing this on the overall journey, How would it help in the Kibana context since the whole journey could contain multiple elements with these test attributes?

setting it as  `beforeAll` will not work for inline scripts.  Plus we will have to import selectors from playwright-chromium directly. Seems hack to me going outside agent package.

`Also, since you are doing this on the overall journey, How would it help in the Kibana context since the whole journey could contain multiple elements with these test attributes?
`
can you please explain this question?

**Comment by shahzad31** [comment]
> @shahzad31 Whats the motivation behind this change? I have couple of questions on this

basic motivation is ability to use `page.getByTestId` selector.

File: __tests__/core/gatherer.test.ts [hunk 2]
```diff
@@ -164,6 +164,41 @@ describe('Gatherer', () => {
     });
   });
 
+  describe.only('Set Test ID Attribute', () => {
```
Comment (vigneshshanmugam) [inline __tests__/core/gatherer.test.ts:167]: can you remove .only?

**Review by vigneshshanmugam — CHANGES_REQUESTED** [review]

File: __tests__/core/gatherer.test.ts (hunk 2 above)
Comment (shahzad31) [inline __tests__/core/gatherer.test.ts:167]:
```suggestion
  describe('Set Test ID Attribute', () => {
```

**Review by vigneshshanmugam — APPROVED** [review]
LGTM

**Pull request merged**